src/praisonai-agents/praisonaiagents/llm/streaming_protocol.py

```
@@ -0,0 +1,387 @@
+    """
+    
+    def can_stream(self, *, tools: Optional[List[Dict[str, Any]]] = None, **kwargs) -> bool:
+        """
+        Anthropic streaming is disabled due to litellm bug.
+        
+        Issue: litellm.acompletion with stream=True returns a ModelResponse (not async generator)
+        for Anthropic in the async path, causing 'async for requires __aiter__' error.
+        """
+        return False  # Disable until litellm bug is fixed
+    
+    def create_stream_unavailable_event(self, reason: str = None, **metadata) -> StreamEvent:
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)